src/main/java/pl/sda/refactoring/service/OrderService.java

```diff
@@ -0,0 +1,156 @@
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static java.math.RoundingMode.HALF_UP;
+import static java.util.UUID.randomUUID;
+
+@Service
+@Transactional
+public class OrderService {
+
+    private final DiscountService discountService;
+    private final CurrencyService currencyService;
+    private final CustomerService customerService;
+    private final OrderRepo orderRepo;
+    private final EmailService emailService;
+    private final List<LocalDate> freeDeliveryDays;
+    private final List<String> emailRecipients;
+
+    public OrderService(@NonNull DiscountService discountService,
+                        @NonNull CurrencyService currencyService,
+                        @NonNull CustomerService customerService,
+                        @NonNull OrderRepo orderRepo,
+                        @NonNull EmailService emailService,
+                        @NonNull OrderSettings settings) {
+        this.discountService = discountService;
+        this.currencyService = currencyService;
+        this.customerService = customerService;
+        this.orderRepo = orderRepo;
+        this.emailService = emailService;
+        this.freeDeliveryDays = settings.discountSettings().freeDeliveryDays();
+        this.emailRecipients = settings.emailRecipients();
+    }
+
+    public UUID makeOrder(@NonNull UUID customerId,
+                          @NonNull List<OrderItem> orderItems,
+                          @NonNull Currency currency,
+                          String coupon) {
+        boolean exists = customerService.exists(customerId);
+        if (exists) {
+            if (!orderItems.isEmpty()) {
+                // exchange currencies in order items
+                List<OrderItem> exchItems = new ArrayList<>();
+                for (var item : orderItems) {
+                    var exchItem = item.toBuilder()
+                        .id(UUID.randomUUID())
+                        .exchPrice(currencyService.exchange(item.price(), item.currency(), currency))
+                        .build();
+                    exchItems.add(exchItem);
+                }
+
+                // calculate delivery
+                var tp = BigDecimal.ZERO;
+                var twInGrams = 0;
+                for (var item : exchItems) {
+                    tp = tp.add(item
+                            .exchPrice()
+                            .multiply(BigDecimal.valueOf(item.quantity())))
+                        .setScale(2, HALF_UP);
+                    switch (item.weightUnit()) {
+                        case GM -> twInGrams += item.weight();
+                        case KG -> twInGrams += item.weight() * 1000;
+                        case LB -> twInGrams += item.weight() * 453.59237;
+                    }
+                }
+                var deliveryPrice = BigDecimal.ZERO;
+                // delivery costs are in USD
```

Comment:
consider adding variable like BASE_CURRENCY instead using comments or using specific currency by default in various methods without explanation